### DISCUSSION TITLE
默认clash mode疑问

### DISCUSSION THREAD

**Discussion by hussion** [kkkgo/PaoPaoGateWay#44] [问和答] [answered]
看了一下源码，ppgw的clash运行模式，tcp和udp好像都是走的tproxy转发，想问下，tcp为何不用redir模式转发呢？以及udp为何不用tun模式？

**Comment by kkkgo** [comment]
所以为什么要用redir模式和tun模式呢？

**Reply by hussion** [reply]
这块知识不是太了解，能否给介绍下？

**Reply by kkkgo** [reply, marked as answer]
如果你说的是redir-host，这个不是fakeip模式，并且存在问题，redir-host功能早已经被clash移除。
如果你说的是iptables的Redirect，设计上并不支持UDP和IPV6；其他平台也忽略了这种方式的文档。
TProxy工作在内核层面性能开销比tun要小。并且tun是clash Pro闭源版本的功能。

**Reply by hussion** [reply]
> 如果你说的是iptables的Redirect，设计上并不支持UDP和IPV6；其他平台也忽略了这种方式的文档。

感谢解答，单拿tcp来讲，iptables Redirect和TProxy区别大吗？

**Reply by kkkgo** [reply]
tproxy是使用 socket marking 机制对数据包进行标记，而不需要修改IP数据包报文，原封不动地据标记进行更灵活的策略路由。而REDIRECT只是简单的修改报文端口重定向。
